docs/Makefile

```diff
@@ -7,13 +47,18 @@ SPHINXBUILD   = sphinx-build
 SOURCEDIR     = .
 BUILDDIR      = _build
 
-# Put it first so that "make" without argument is like "make help".
-help:
-	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+# Build the "html" Sphinx target which is what "Read the Docs" wants.  There are
+# other supported targets one could use; see `$(SPHINXBUILD) -M help` for a
+# complete list if interested.
+.PHONY: html
+html:
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS)
 
-.PHONY: help Makefile
+.PHONY: clean
+clean:
+	rm -f *~ *.fdb_latexmk *.fls *.toc *.aux *.log *.out *.xdv
+	rm -rf $(BUILDDIR)
 
-# Catch-all target: route all unknown targets to Sphinx using the new
-# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-%: Makefile
-	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```

Comment:
I'm feeling kind of sad that we're losing the ability to pass unknown targets on to `$SPHINXBUILD`.

If you change the `html:` target to `%:`, does that really cause a problem?  (According to the docs, it shouldn't interfere with the `%.html`, `%.tex`, and `%.pdf` targets because they come before it in the makefile, and it shouldn't be selected over the `push-*` targets because they are more specific.)

(I can imagine how it could cause a problem for the `clean` or `veryclean` targets (although it _shouldn't_), but that would be easy to address by placing their definitions before the catch-all one.)

Reply:
Yes, you get whacky output try to build the other existing targets.  We'll resolve this later once we decide what documentation build system we should settle on for the future.

I suspect it will be `Sphinx`, but I did not want to do all sorts of whole-sale changes in this PR.  This PR is trying to just bring directories together.